dependency_support/verilator/private/verilator_utils.bzl

```diff
@@ -52,85 +52,78 @@ verilator_astgen = rule(
-    ])
+    args.add("--yacc", bison_toolchain.bison_tool.executable)
+    args.add("-d")
+    args.add("-v")
+    args.add("-o", ctx.outputs.out_src)
+    args.add(ctx.file.yacc_src)
+
+    outputs = [
+        ctx.outputs.out_src,
+        ctx.outputs.out_hdr,
+    ]
+
+    tools = depset([ctx.file.bisonpre], transitive = [bison_toolchain.all_files])
+
+    bison_env = correct_bison_env_for_action(
+        env = bison_toolchain.bison_env,
+        bison = bison_toolchain.bison_tool.executable,
+    )
 
-    envs = {
-        "BISON_PKGDATADIR": locate_bison_data_root(ctx.files.bison_data),
-    }
-    envs.update({
-        # Expand $(location) / $(locations) in the values.
-        k: ctx.expand_location(v, data) if _is_expandable(v) else v
-        for k, v in ctx.attr.env.items()
-    })
     ctx.actions.run(
-        outputs = ctx.outputs.outs,
-        inputs = ctx.files.srcs,
-        tools = [ctx.file.bisonpre] + ctx.files.tools + ctx.files.bison_data,
+        outputs = outputs,
+        inputs = [ctx.file.yacc_src],
+        tools = tools,
         executable = ctx.executable._process_wrapper,
         arguments = [args],
         mnemonic = "VerilatorBisonPre",
         use_default_shell_env = False,
-        env = envs,
+        env = bison_env,
     )
+
     return DefaultInfo(
-        files = depset(ctx.outputs.outs),
-        runfiles = ctx.runfiles(files = ctx.outputs.outs),
+        files = depset(outputs),
+        runfiles = ctx.runfiles(files = outputs),
     )
 
 verilator_bisonpre = rule(
     doc = "Run Verilator's `bisonpre` tool and collect the requested outputs.",
     implementation = _verilator_bisonpre_impl,
     attrs = {
-        "args": attr.string_list(
-            doc = "Command line arguments of the `bisonpre`",
-        ),
-        "bison_data": attr.label(
-            doc = "Runtime data for GNU Bison.",
-            allow_files = True,
-            mandatory = True,
-        ),
         "bisonpre": attr.label(
             doc = "The path to the `bisonpre` tool.",
             allow_single_file = True,
             mandatory = True,
             cfg = "exec",
         ),
-        "env": attr.string_dict(
-            doc = "Environment variables of the action.",
-        ),
-        "outs": attr.output_list(
+        "out_hdr": attr.output(
             mandatory = True,
             doc = "Output files generated by the action.",
         ),
-        "srcs": attr.label_list(
-            allow_files = True,
-            doc = "Additional inputs of the action.",
+        "out_src": attr.output(
+            mandatory = True,
+            doc = "Output files generated by the action.",
         ),
-        "tools": attr.label_list(
-            allow_files = True,
-            cfg = "exec",
-            doc = "Additional tools of the action.",
+        "yacc_src": attr.label(
+            doc = "The yacc file to run on.",
+            allow_single_file = True,
+            mandatory = True,
```

Comment:
This seems like a backwards incompatible change. Would this cause errors for downstream users?

Reply:
To my knowledge this rule is only used internally so the change should not have an impact. Their existence in a `private` directory was to discourage users from loading them (as buildifier will complain with `bzl-visibility`).